src/core/thread/discover_scanner.cpp

```diff
@@ -144,6 +157,22 @@ otError DiscoverScanner::Discover(const Mac::ChannelMask &aScanChannels,
     return error;
 }
 
+otError DiscoverScanner::SetJoinerAdvertisement(uint32_t aOui, const uint8_t *aAdvData, uint8_t aAdvDataLength)
+{
+    otError error = OT_ERROR_NONE;
+
+    VerifyOrExit((aAdvData != nullptr) && (aAdvDataLength != 0) && (aAdvDataLength <= kMaxLength),
+                 error = OT_ERROR_INVALID_ARGS);
+
+    mOui                    = aOui;
+    mAdvData                = aAdvData;
```

Comment:
Should `memcpy` the data to `mAdvData` instead of just copying the pointer? 
`SetJoinerAdvertisement` is called by `otThreadSetJoinerAdvertisement` API. `aAdvData` might be a pointer to stack data.

Reply:
Agree, fixed.